src/wallet/wallet.cpp

```diff
@@ -5295,7 +5295,7 @@ bool CWallet::AutoBackupWallet(const fs::path& wallet_path, std::string& strBack
     }
 
     // Create backup of the ...
-    std::string dateTimeStr = DateTimeStrFormat(".%Y-%m-%d-%H-%M", GetTime());
+    std::string dateTimeStr = FormatISO8601DateTime(GetTime());
```

Comment:
Should be _at least_
```suggestion
    std::string dateTimeStr = "." + FormatISO8601DateTime(GetTime());
```
for backup maintenance (i.e. keep 10 recent backups only) to work correctly. But this would still break the logic preventing backups to appear too fast (see a warning below). What you actually need to preserve the way it works currently is smth like
```suggestion
    struct tm ts;
    time_t time_val = GetTime();
    gmtime_r(&time_val, &ts);
    std::string dateTimeStr = strprintf(".%04i-%02i-%02i-%02i-%02i", ts.tm_year + 1900, ts.tm_mon + 1, ts.tm_mday, ts.tm_hour, ts.tm_min);
```